docs/admin/node-problem.md

```diff
@@ -0,0 +1,199 @@
+spec:
+  template:
+    metadata:
+      labels:
+        k8s-app: node-problem-detector
+        version: v0.1
+        kubernetes.io/cluster-service: "true"
+    spec:
+      hostNetwork: true
+      containers:
+      - name: node-problem-detector
+        image: gcr.io/google_containers/node-problem-detector:v0.1
+        securityContext:
+          privileged: true
+        resources:
+          limits:
+            cpu: "200m"
+            memory: "100Mi"
+          requests:
+            cpu: "20m"
+            memory: "20Mi"
+        volumeMounts:
+        - name: log
+          mountPath: /log
+          readOnly: true
+        - name: config # Overwrite the config/ directory with ConfigMap volume
+          mountPath: /config
+          readOnly: true
+      volumes:
+      - name: log
+        hostPath:
+          path: /var/log/
+      - name: config # Define ConfigMap volume
+        configMap:
+          name: node-problem-detector-config
+```
+* **Step 4:** Re-create the node problem detector with the new yaml file:
+
+```shell
+kubectl delete -f node-problem-detector.yaml # If you have a node-problem-detector running
+kubectl create -f node-problem-detector.yaml
+```
+***Notice that this approach only applies to node problem detector started with `kubectl`.***
+
+For node problem detector running as cluster addon, because addon manager doesn't support
+ConfigMap, configuration overwriting is not supported now.
+
+## Kernel Monitor
+*Kernel Monitor* is a problem daemon in node problem detector. It monitors kernel log
+and detects known kernel issues following predefined rules.
+
+The Kernel Monitor matches kernel issues according to a set of predefined rule list in
+[`config/kernel-monitor.json`](https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json).
+The rule list is extensible, you can always extend it by [overwriting the
+configuration](/docs/admin/node-problem/#overwrite-the-configuration).
+
+### Add New Node Conditions
+To support new node conditions, you can extend the `conditions` field in
+`config/kernel-monitor.json` with new condition definition:
+
+```json
+{
+  "type": "NodeConditionType",
+  "reason": "CamelCaseDefaultNodeConditionReason",
+  "message": "arbitrary default node condition message"
+}
+```
+
+### Add New Issues
+To support new issues, you can extend the `rules` field in `config/kernel-monitor.json`
+with new rule definition:
+
+```json
+{
+  "type": "temporary/permanent",
+  "condition": "NodeConditionOfPermanentIssue",
+  "reason": "CamelCaseShortReason",
+  "message": "regexp matching the issue in the kernel log"
+}
+```
+
+### Change Log Path
+Kernel log in different OS distros may locate in different path. The `log`
+field in `config/kernel-monitor.json` is the log path inside the container.
+You can always config it to match your OS distro.
+
+***Notice that Kernel Monitor doesn't support log tools like journald for now.
```

Comment:
Known issues should go near the top so users don't try to get it working and fail.